packages/block-library/src/image/image-editing/use-transform-image.js

```diff
@@ -12,6 +13,12 @@ function useTransformState( { url, naturalWidth, naturalHeight } ) {
 	const [ aspect, setAspect ] = useState();
 	const [ defaultAspect, setDefaultAspect ] = useState();
 
+	const imageCrossOrigin = useSelect( ( select ) => {
+		const value = select( 'core/block-editor' ).getSettings()
```

Comment:
Could we maybe use a filter in JS here, instead of using block settings?

Reply:
That certainly seems technically possible. Do you think that would be better?

This seems to me like site configuration that the backend is able to make decisions about. It doesn't seem like the kind of thing the client side needs to be deciding at runtime.

Reply:
I'm not familiar with WP js filters, would it be possible to pick up a value set from the server-side / PHP with it? Could you provide an example? Thanks!

Reply:
Filter seems to make sense to me, as it's more aligned with WordPress's typical way of extending/customizing things, whereas `block-settings` could be deemed an internal carrier of information (and thus, and implementation detail that might change).

On JS vs server-side/PHP filter: Agree that 

> It doesn't seem like the kind of thing the client side needs to be deciding at runtime.

A point could be made however that making it a JS filter makes some sense, as it "cuts out the middle man": It's a customization that needs to happen on the client side, so it's arguable that it doesn't have to be Gutenberg's responsibility to take on the chore of communicating it from the server to the client. (As long as that can be done easily enough by whatever plugin/extensions customizes this.)

(I don't feel very strongly about the JS vs PHP filter though.)

Reply:
I've been reflecting on this and there _may_ be a good reason to handle it in JavaScript.

The case we're considering is when we're on our site (example.com) and we want to use images from our CDN on a different domain (example.net). That's easy to cover in this case because we understand usage at this time.

However, we only want to set `crossOrigin` requests for our known domains and not for arbitrary ones. We might want to `use-credentials` for our known `example.net`, but this hook could be used with other domains where we wouldn't want to send our credentials. We'd need to make this decision at runtime, or have some more complex data to describe crossOrigin based on domain:

```js
addFilter( 'imageTransform.crossOrigin', 'my-filter', ( crossOrigin, imgSrc ) => {
  if ( imgSrc.startsWith( 'https://example.net' ) ) {
    return 'use-credentials';
  }
  return crossOrigin;
} )
```

To acheive something similar via block editor settings we'd need something like the following:

```php
function configure_block_settings_media_crossorigin( $settings ) {
	$settings['mediaCrossOriginDomains'] = array(
		'example.net' => 'use-credentials',
	);
	return $settings;
};
add_filter( 'block_editor_settings', 'configure_block_settings_media_crossorigin' );
```

Reply:
I've changed this to use a filter, like you suggested above, @sirreal. There's a nice side-effect in that it requires much less code :).

>  but this hook could be used with other domains where we wouldn't want to send our credentials. We'd need to make this decision at runtime, or have some more complex data to describe crossOrigin based on domain:

I'm not sure I fully understood what you meant here though. In the filter snippet above you're returning `use-credentials` if the `img` `src` matches a certain domain, and then `crossOrigin` as is, if not. At first sight, it didn't seem to match the logic we had before, so I instead tested with a filter that looks like this:

```
	addFilter(
		'gutenberg.imageTransform.crossOrigin',
		'gutenberg/imageTransform',
		( imgSrc ) => {
			if (
				// Only set crossOrigin for images served by us
				imgSrc.match( /https:\/\/.*\.files.adomain.com\/.*/ )
			) {
				return attrs.image_transform_cors_image_cross_origin; // this is calculated in the PHP backend and passed here using `wp_localize_script`. It returns `anonymous` or `use-credentials`.
			}
		}
	);
```

Let me know if this is what you had in mind. Thanks!